### DISCUSSION TITLE
rpmdb - flag about the package installation reason

### DISCUSSION THREAD

**Discussion by scarabeusiv** [rpm-software-management/rpm#2767] [General]
When installing anything on the system all packages are treated as equals, which in turn makes it pretty hard to interpret wether user decided to install package A or if it was pulled and installed only as dependency.

It would be nice to store a marker for each package to figure out where it came from and then let the libsolv and others to allow clean and easy way how to remove unneeded dependencies.

Most of the package managers (zypp, dnf) kinda support this idea but I think this is safest in RPM directly as user still can use rpm calls to install packages. It would be more scatchy if it would be stored by dnf/zypper somewhere compared to direct rpm storage.

**Comment by Conan-Kudo** [comment]
The idea has been bandied about before to migrate the "reason" info stored in yumdb (now using SWDB in newer versions of DNF). I don't know what really holds up a potential implementation of this in the rpmdb.

@dmach, @ffesti, @pmatilai: what do you guys think?

**Comment by pmatilai** [comment]
Commit b3fcaba3816aa00f3360efe728dafa72e394f65b added a tag reservation basically just for this purpose but the implementation got stalled with various underlying defiences, side-issues and API considerations.

**Comment by pmatilai** [comment]
This needs an rpmte/rpmts API rework to be sane, currently its ridiculously cumbersome to find the element you just added in order to add some extra data to the element. There are several use-cases that would benefit from such work though.
